src/Microsoft.AspNet.Mvc/MvcServices.cs

```diff
@@ -151,8 +151,10 @@ public static IEnumerable<IServiceDescriptor> GetDefaultServices(IConfiguration
             // IActionBindingContextProvider. Therefore it too is scoped.
             yield return describe.Transient<IHtmlGenerator, DefaultHtmlGenerator>();
 
-            yield return describe.Transient<IViewComponentSelector, DefaultViewComponentSelector>();
+            // These do caching so they should stay singleton
```

Comment:
No, the activator is like our other activators - it caches the delegates needed to set properties. 

Separate caches is better in this case because they are pay-for-play. If you never invoke a specific component, we never build an activation cache for it, which is great. However we need to do discovery (and its caching) if you use any VC.